lib/logstash/outputs/elastic_app_search.rb

```diff
@@ -7,14 +7,26 @@ class LogStash::Outputs::ElasticAppSearch < LogStash::Outputs::Base
   config_name "elastic_app_search"
 
   config :engine, :validate => :string, :required => true
-  config :host, :validate => :string, :required => true
+  config :host, :validate => :string
+  config :url, :validate => :string
   config :api_key, :validate => :password, :required => true
   config :timestamp_destination, :validate => :string
   config :document_id, :validate => :string
+  config :path, :validate => :string, :default => "/api/as/v1/"
 
   public
   def register
-    @client = Client.new(@host, @api_key.value)
+    if @host.nil? && @url.nil?
+      raise ::LogStash::ConfigurationError.new("Please specify either \"url\" (for self-managed) or \"host\" (for SaaS).")
+    elsif @host && @url
+      raise ::LogStash::ConfigurationError.new("Both \"url\" or \"host\" can't be set simultaneously. Please specify either \"url\" (for self-managed) or \"host\" (for SaaS).")
+    elsif @host && @path
```

Comment:
since `path` has a default value, by the time we get here `@path` will be populated whether or not it was specified.

Reply:
fixed by looking at the `original_params` hash. Also now with tests to confirm exceptions are raised (or not) under the right conditions.